code/wdb/writedown.q

```diff
@@ -0,0 +1,23 @@
+/-endofperiod function to savedown by integer
+endofperiod:{[currp;nextp;data]
+	.lg.o[`endofperiod;"Received endofperiod. currentperiod, nextperiod and data are ",(string currp),", ", (string nextp),", ", .Q.s1 data];
+        /- Obtain handle of any other running wdb's
+	h:exec w from .servers.SERVERS where proctype=`wdb,not w=0N;
+        /-Create a list of start times of wdb's found from above
+	times:@[;".proc.starttimeUTC";()]each h;
+        /-If we are the new process, exit function, do not want to writedown
+	if[.proc.starttimeUTC >max times;
+                /-Setting variables so wdb can become the active wdb for this new period
+		@[`.;`upd;:;insert];
+		.wdb.currentpartition:`long$nextp;
+		:()];
+        /-We must be old process so must writedown
```

Comment:
this comment doesn't directly explain the following line, might be worth adding something here to indicate that first step is closing TP subscriptions